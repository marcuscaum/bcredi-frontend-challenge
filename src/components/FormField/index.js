import React from 'react';
import PropTypes from 'prop-types';

import { FormField, Label } from './index.styled';

export const FormFieldComponent = ({ label, error, children }) => (
  <FormField error={error}>
    {label && <Label>{label}</Label>}
    {children}
    {error && <span>{error}</span>}
  </FormField>
);

FormFieldComponent.propTypes = {
  error: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
  label: PropTypes.string,
  children: PropTypes.element.isRequired,
};

FormFieldComponent.defaultProps = {
  error: null,
  label: null,
};

export default FormFieldComponent;
