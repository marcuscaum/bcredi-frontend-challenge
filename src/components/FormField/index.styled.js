import styled from 'styled-components';

export const FormField = styled.div`
  display: flex;
  flex-direction: column;
  margin-bottom: 35px;
  min-width: 0;

  input {
    padding: 14px;
    font-size: 16px;
    border-radius: 3px;
    border: 1px solid
      ${({ error, theme: { destructive, ultraLightGray } }) =>
        error ? destructive : ultraLightGray};

    &::placeholder {
      color: #cccccc;
    }
  }

  span {
    color: ${({ theme: { destructive } }) => destructive};
    line-height: 1.29;
    font-size: 14px;
    margin-top: 5px;
  }
`;

export const Label = styled.label`
  font-size: 14px;
  margin-bottom: 7px;
  color: ${({ theme: { lightGray } }) => lightGray};
`;
