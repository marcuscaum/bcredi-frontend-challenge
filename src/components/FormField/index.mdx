---
name: FormField
---


import { Playground, PropsTable } from 'docz'
import FormField from './index'

# FormField

<PropsTable of={FormField} />

## Basic usage

<Playground>
  <FormField 
    label="Exemplo de label"
    error="Digite um email vÃ¡lido"
  >
    <input type="text" placeholder="Digite aqui seu email" />
  </FormField>
</Playground>