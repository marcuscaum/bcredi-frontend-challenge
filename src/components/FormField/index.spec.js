import React from 'react';
import { shallow } from 'enzyme';

import FormField from '.';

const setup = (props = {}) =>
  shallow(
    <FormField {...props}>
      <div className="to-not-trigger-proptypes" />
    </FormField>
  );

describe('[Component] FormField', () => {
  it('should render the component', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
