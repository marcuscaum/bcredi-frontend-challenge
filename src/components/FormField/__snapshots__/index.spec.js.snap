// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] FormField should render the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    error={null}
    label={null}
>
    <div
        className="to-not-trigger-proptypes"
    />
</Unknown>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        null,
        <div
          className="to-not-trigger-proptypes"
/>,
        null,
      ],
      "error": null,
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "to-not-trigger-proptypes",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      null,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          null,
          <div
            className="to-not-trigger-proptypes"
/>,
          null,
        ],
        "error": null,
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "to-not-trigger-proptypes",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        null,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
